config/worker.js

```diff
@@ -0,0 +1,146 @@
+const CLIENT_ID = '6caaf5d0a58e47868553160955e5b8e4'
+const CLIENT_SECRET = 'REDACTED'
+const REDIRECT_URL = '/spotifycb'
+const WORKER_URL = 'https://duoco.cawman.workers.dev'
+const CALLBACK_URL = 'file://callback'
+
+const refreshtokensMap = new Map()
+
+/**
+ * Example of how router can be used in an application
+ *  */
+addEventListener('fetch', event => {
+    event.respondWith(handleRequest(event.request))
+})
+
+
+async function handleRequest(request) {
+    let requestURL = new URL(request.url)
+    
+    switch(requestURL.pathname) {  
+        case '/spotifylogin': {
+            let scopes = 'user-read-currently-playing'
+            //let referrer = requestURL.searchParams.get('ref')
+            let url = 'https://accounts.spotify.com/authorize' +
+            '?response_type=code' +
+            '&client_id=' + CLIENT_ID +
+            (scopes ? '&scope=' + encodeURIComponent(scopes) : '') +
+            '&redirect_uri=' + encodeURIComponent(CALLBACK_URL)// + '&state=' + referrer
+            return new Response(null, {
+                status: 302,
+                headers: {
+                    'location': url
+                }
+            })
+        }
+        case '/refreshtoken': {
+            let code = request.headers.get('Authorization')
+            if(request.method == 'OPTIONS') {
+                return new Response(null, {
+                    status: 204,
+                    headers: {
+                        'Access-Control-Allow-Origin': '*',
+                        'Access-Control-Allow-Methods': 'GET',
+                        'Access-Control-Allow-Headers': 'Authorization'
+                    }
+                })
+            }
+            let resp = await fetch('https://accounts.spotify.com/api/token', {
```

Comment:
Told you already, you shouldn't be hardcoding the API url. It should be in the config file.